internal/anchore/anchoreclient.go

```diff
@@ -0,0 +1,250 @@
+package anchore
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/anchore/k8s-inventory/internal/config"
+	"github.com/anchore/k8s-inventory/internal/log"
+	"github.com/anchore/k8s-inventory/internal/tracker"
+	"github.com/h2non/gock"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"syscall"
+	"time"
+)
+
+type ControllerErrorDetails struct {
+	Type   string `json:"type"`
+	Title  string `json:"title"`
+	Detail string `json:"detail"`
+	Status int    `json:"status"`
+}
+
+type APIErrorDetails struct {
+	Message  string                 `json:"message"`
+	Detail   map[string]interface{} `json:"detail"`
+	HTTPCode int                    `json:"httpcode"`
+}
+
+type APIClientError struct {
+	HTTPStatusCode         int
+	Message                string
+	Path                   string
+	Method                 string
+	Body                   *[]byte
+	APIErrorDetails        *APIErrorDetails
+	ControllerErrorDetails *ControllerErrorDetails
+}
+
+func (e *APIClientError) Error() string {
+	return fmt.Sprintf("API errorMsg(%d): %s Path: %q %v %v", e.HTTPStatusCode, e.Message, e.Path,
+		e.APIErrorDetails, e.ControllerErrorDetails)
+}
+
+func Post(requestBody []byte, id string, path string, anchoreDetails config.AnchoreInfo, operation string) (*[]byte, error) {
```

Comment:
If we're adding a generic anchoreclient here we should probably modify pkg/reporter to use it

Reply:
Yes that makes sense. I would prefer to have a separate refactoring PR for that though in order to reduce risk of regressions.